src/D2L.CodeStyle.Analyzers/ApiUsage/Serialization/SerializerAttributeAnalyzer.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System.Collections.Immutable;
+using System.Linq;
+using D2L.CodeStyle.Analyzers.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace D2L.CodeStyle.Analyzers.ApiUsage.Serialization {
+
+	[DiagnosticAnalyzer( LanguageNames.CSharp )]
+	internal sealed class SerializerAttributeAnalyzer : DiagnosticAnalyzer {
+
+		private static readonly SymbolDisplayFormat TypeDisplayFormat = new SymbolDisplayFormat(
+				memberOptions: SymbolDisplayMemberOptions.IncludeContainingType,
+				localOptions: SymbolDisplayLocalOptions.IncludeType,
+				typeQualificationStyle: SymbolDisplayTypeQualificationStyle.NameAndContainingTypesAndNamespaces
+			);
+
+		private const string SerializerAttributeFullName = "D2L.LP.Serialization.SerializerAttribute";
+		private const string ITrySerializerFullName = "D2L.LP.Serialization.ITrySerializer";
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(
+			Diagnostics.InvalidSerializerType
+		);
+
+		public override void Initialize( AnalysisContext context ) {
+			context.EnableConcurrentExecution();
+			context.RegisterCompilationStartAction( RegisterScopeBuilderAnalyzer );
+		}
+
+		private void RegisterScopeBuilderAnalyzer( CompilationStartAnalysisContext context ) {
+
+			Compilation comp = context.Compilation;
+			if( !comp.TryGetTypeByMetadataName( SerializerAttributeFullName, out INamedTypeSymbol serializerAttributeType ) ) {
+				return;
+			}
+			if( !comp.TryGetTypeByMetadataName( ITrySerializerFullName, out INamedTypeSymbol trySerializerType ) ) {
+				return;
+			}
+
+			context.RegisterSyntaxNodeAction(
+					c => AnalyzeAttributeSyntax(
+						c,
+						(AttributeSyntax)c.Node,
+						serializerAttributeType,
+						trySerializerType
+					),
+					SyntaxKind.Attribute
+				);
+		}
+
+		private void AnalyzeAttributeSyntax(
+				SyntaxNodeAnalysisContext context,
+				AttributeSyntax attributeSyntax,
+				INamedTypeSymbol serializerAttributeType,
+				INamedTypeSymbol trySerializerType
+			) {
+
+			SemanticModel model = context.SemanticModel;
+			if( !IsAttributeOfType( model, attributeSyntax, serializerAttributeType ) ) {
+				return;
+			}
+
+			AttributeArgumentSyntax typeArgumentSyntax = attributeSyntax.ArgumentList.Arguments[ 0 ];

```

Comment:
I believe this would crash the analyzer if an attempt was made to compile without specifying args (which would also fail to compile, at least with the current definition, but I believe analyzers will run first).

Consider handling this with some null propagation and `.FirstOrDefault()`.

```suggestion
			AttributeArgumentSyntax typeArgumentSyntax = attributeSyntax.ArgumentList?.Arguments?.FirstOrDefault();
			if( typeArgumentSyntax is null ) {
			    // compilation will fail
			    return;
			}
```

(and test cases for `[Serializer]` and `[Serializer()]`)

Reply:
Oops, this was already resolved.